diff --git a/elixir/config/config.exs b/elixir/config/config.exs
index 0f31067..068d2e1 100644
--- a/elixir/config/config.exs
+++ b/elixir/config/config.exs
@@ -18,8 +18,6 @@ config :portal, generators: [binary_id: true]
 config :portal, sql_sandbox: false
 config :portal, replica_repo: Portal.Repo.Replica

-# Don't run manual migrations by default
-config :portal, run_manual_migrations: false

 config :portal, Portal.Repo,
   hostname: "localhost",
